h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -1173,21 +1248,52 @@ private Chunk chk_xnew(Chunk[] chks, int c) {
         // Categorical columns
         for (int j = 0; j < _ncats; j++) {
           if (Double.isNaN(a[j])) continue;   // Skip missing observations in row
-          double[] xy = ArrayUtils.multVecArr(xnew, _yt.getCatBlock(j));
-          _loss +=  _lossFunc[j].mloss(xy, (int) a[j]);
+//          double[] xy = ArrayUtils.multVecArr(xnew, _yt.getCatBlock(j));
+//          _loss +=  _lossFunc[j].mloss(xy, (int) a[j]);
+
+          double[] txy = multVecArrFast(xnew, _yt, j);
+          _loss +=  _lossFunc[j].mloss(txy, (int) a[j]);
```

Comment:
Same comment as above applies here as well: right now `txy` is re-allocated on every iteration (and for every categorical). However the `xy` above can be reused here, provided that you add an extra parameter "effective length of `xy`" to the signature of `multVecArrFast`.